internal/resources/fetcher.go

```diff
@@ -110,3 +119,59 @@ func addPolicyResources(resources map[schema.GroupVersionResource][]policiesv1.P
 		}
 	}
 }
+
+func getPolicyServerByName(ctx context.Context, policyServerName string, dynamicClient *dynamic.Interface) (*policiesv1.PolicyServer, error) {
+	resourceID := schema.GroupVersionResource{
+		Group:    "policies.kubewarden.io",
+		Version:  "v1",
+		Resource: "policyservers",
+	}
+	resourceObj, err := (*dynamicClient).Resource(resourceID).Get(ctx, policyServerName, metav1.GetOptions{})
```

Comment:
I think is doable. But I decided to use the `dynamicClient` because is the client already in use in this fetcher object. The fetcher used to get the policies uses the client (`client.Client`) that you are talking about. But I do not know the  root cause to use the dynamic client here.

The impression that I had reading the code and [comments](https://github.com/kubewarden/audit-scanner/blob/95f3b8d4d47dfe96d181ecf008521a6ab5007ba4/internal/resources/fetcher.go#L16) is to avoid trying to get a resource that is, for some reason, not supported by the `client.Client`. Taking a look in the [docs](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#New) of the `client.Client` it seems it is able to handle almost [all](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Object) of the resources. But it can still fail in some weird cases. So, I guess we can use the same client everywhere. But I need to double check if there is some unusual resource that we must support that could cause issues if we move to `client.Client`.

Reply:
@raulcabello can you provide some background information about why we are using this dynamic client?

Reply:
From what I understand from https://pkg.go.dev/k8s.io/client-go@v0.26.1/examples/dynamic-create-update-delete-deployment#readme-typed-vs-dynamic, the dynamic client uses `unstructured.Unstructured` to represent objects from the API server, which provides us forward compat with our CRDs, and being able to handle reports of objects that the client hasn't been made aware of (CRDs in the cluster, etc).

Reply:
Considering that using typed resources may requires a recompilation of the code to be able to run with the lastest version. I would say to continue to use the dynamic client and write a good set of test to ensure that everything is working as expected.

Reply:
Thanks for the explanation. I'm totally fine with the usage of the dynamic client